TreeView: add the "getting started" article (#2591)
DevExpress/devextreme-documentation: concepts/05 UI Components/TreeView/00 Getting Started with TreeView/50 Select Records.md

> To enable node selection, set the [selectByClick](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick) property to **true**. If you want to add checkboxes for each node, set the [showCheckBoxesMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#showCheckBoxesMode) property to *"normal"* or *"selectAll"*. The latter mode additionally displays the "Select All" checkbox at the top of the TreeView.

Reviewer: Rename the article to Select Nodes.

Why don't you use already written and verified texts?

```suggestion
To select a node, users can click a checkbox next to it. Set the [showCheckBoxesMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#showCheckBoxesMode) to *"normal"* or *"selectAll"* to display node checkboxes. The *"selectAll"* mode also enables a checkbox that selects all nodes simultaneously. If [selectByClick](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick) is enabled, users can click nodes to select them.
```

> The default selection mode is multiple. As an alternative, you can set the [selectionMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/selectionMode) property to *"single"* as in this demo.

Reviewer: ```suggestion
The TreeView supports multiple (default) and single [selectionMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/selectionMode). The code below shows how to enable the single mode.
```

> To obtain the selected node's data, use the [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#onSelectionChanged) function. In the code below, this function displays the selected product and its price:

Reviewer: Because I used "the code below" in the previous suggestion:

```suggestion
To obtain the selected node's data, use the [onItemSelectionChanged](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#onItemSelectionChanged) function. In the following code, this function displays the selected product and its price:
```

    });

    <!-- tab: index.html -->
    <!DOCTYPE html>
    <html>
        <head>
            <!-- ... -->
        </head>
        <body class="dx-viewport">
            <div id="treeView"></div>
            <div id="product-details" class="hidden">
                <img src="" />
                <div class="name"></div>
                <div class="price"></div>
            </div>
        </body>
    </html>

    <!-- tab: index.css -->
    #treeView, #product-details {
        display: inline-block;
        width: 300px;
    }

    #product-details {
        vertical-align: top;
        width: 400px;
        height: 420px;
        margin-left: 20px;
    }

    #product-details > img {
        border: none;
        height: 300px;
        width: 400px;
    }

    #product-details > .name {
        text-align: center;
        font-size: 20px;
    }

    #product-details > .price {
        text-align: center;
        font-size: 24px;
    }

    .hidden {
        visibility: hidden;
    }

##### Angular

    <!-- tab: app.component.html -->
    <dx-tree-view ...
        selectionMode="single"
        selectByClick=true
        (onItemSelectionChanged)="selectProduct($event)"

>     >

Reviewer: ```suggestion
        selectByClick="true"
        (onItemSelectionChanged)="selectProduct($event)">
```

            vertical-align: top;
            width: 400px;
            height: 420px;
            margin-left: 20px;
        }

        #product-details > img {
            border: none;
            height: 300px;
            width: 400px;
        }

        #product-details > .name {
            text-align: center;
            font-size: 20px;
        }

        #product-details > .price {
            text-align: center;
            font-size: 24px;
        }
    </style>

##### React

    <!-- tab: App.js -->
    // ...
    function App() {
        const [selectedNode, setSelectedNode] = useState(products[0]);
        const selectProduct = (e) => {
            setSelectedNode(e.itemData);
        }
        // ...
        return (
            <div>
                <TreeView ...
                    selectionMode="single"
                    selectByClick={true}
                    onItemSelectionChanged={selectProduct}
                />
                {
                    selectedNode.price &&
                    <div id="product-details">
                    <img src={selectedNode.image}/>
                    <div className="name">{selectedNode.name}</div>

>                     <div className="price">{`$${ selectedNode.price}`}</div>

Reviewer: Alignment is broken.

To enable node selection, you can use the following properties:

> * [selectByClick](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick)

Reviewer: ```suggestion
* [selectByClick](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick)        
```

Enables selection by click.

> * [showCheckBoxesMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#showCheckBoxesMode)

Reviewer: ```suggestion
* [showCheckBoxesMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#showCheckBoxesMode)        
```

                }
            }
        });
    });

    <!-- tab: index.html -->
    <!DOCTYPE html>
    <html>
        <head>
            <!-- ... -->
        </head>
        <body class="dx-viewport">
            <div id="treeView"></div>
            <div id="product-details" class="hidden">
                <img src="" />
                <div class="name"></div>
                <div class="price"></div>
            </div>
        </body>
    </html>

    <!-- tab: index.css -->
    #treeView, #product-details {
        display: inline-block;
        width: 300px;
    }

    #product-details {
        vertical-align: top;
        width: 400px;
        height: 420px;
        margin-left: 20px;
    }

    #product-details > img {
        border: none;
        height: 300px;
        width: 400px;
    }

    #product-details > .name {
        text-align: center;
        font-size: 20px;
    }

    #product-details > .price {
        text-align: center;
        font-size: 24px;
    }

    .hidden {
        visibility: hidden;
    }

##### Angular

    <!-- tab: app.component.html -->
    <dx-tree-view ...
        selectionMode="single"

>         selectByClick="true"

Reviewer: ```suggestion
        [selectByClick]="true"
```

        <img [src]="currentProduct.image" />
        <div class="name">{{currentProduct.name}}</div>
        <div class="price">${{ currentProduct.price }}</div>
    </div>

    <!-- tab: app.component.ts -->
    // ...
    export class AppComponent {
        // ...
        currentProduct: Product;

        constructor(service: ProductsService) {
            // ...
            this.currentProduct = this.products[0];
        } 

        selectProduct(e) {
            this.currentProduct = e.itemData;
        }
    }

    <!-- tab: app.component.css -->
    #treeView, #product-details {
        display: inline-block;
        width: 300px;
    }

    #product-details {
        vertical-align: top;
        width: 400px;
        height: 420px;
        margin-left: 20px;
    }

    #product-details > img {
        border: none;
        height: 300px;
        width: 400px;
    }

    #product-details > .name {
        text-align: center;
        font-size: 20px;
    }

    #product-details > .price {
        text-align: center;
        font-size: 24px;
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div>
            <DxTreeView ...

>                 selectionMode="single"

Reviewer: kebab-case

                :select-by-click="true"

>                 @item-selection-changed="selectProduct" 
>             />

Reviewer: ```suggestion
                @item-selection-changed="selectProduct">
                <!-- ... -->
            </DxTreeView>
```

To enable node selection, you can use the following properties:

* [selectByClick](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick)           

> Enables selection by click.

Reviewer 2: ```suggestion
Enables selection by a click.
```

* [showCheckBoxesMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#showCheckBoxesMode)           

> Adds checkboxes. Set this property to *"normal"* or *"selectAll"*. The latter additionally displays the "Select All" checkbox at the top of the TreeView.

Reviewer 2: From what I understand, the checkboxes are disabled but you can set the mentioned property to `normal` or `selectAll` to enable them. I understand that you may want to start the line with "Adds checkboxes" to maintain the pattern introduced in the previous list item. However, this variant looks more natural to me:

```suggestion
Set this property to *"normal"* or *"selectAll"* to add checkboxes. The latter option additionally displays the "Select All" checkbox at the top of the TreeView.
```

> The TreeView supports multiple (default) and single [selectionMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/selectionMode). The code below shows how to enable the single mode and selection by click.

Reviewer 2: Both this and the following paragraphs are written as if there were two separate code examples to back them. However, these paragraphs refer to the same code example. I guess you should join the API descriptions together and only then describe what exactly the following code example does.

> how to enable the single mode and selection by click

I strongly urge you to rewrite this phrase because it does not look good. First, you do not enable single mode, you enable single-node selection mode. Second, you try to mix both the selection mode and selection by a click in a single phrase. This just does not work. Third, do not use articles with "mode".

> To obtain the selected node's data, use the [onItemSelectionChanged](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#onItemSelectionChanged) function. In the following code, this function displays the selected product, its picture, and price:

Reviewer 2: Where does the `onItemSelectionChanged` function display the selected product, its picture, and price?

        <div class="name">{{currentProduct.name}}</div>
        <div class="price">${{ currentProduct.price }}</div>
    </div>

    <!-- tab: app.component.ts -->
    // ...
    export class AppComponent {
        // ...
        currentProduct: Product;

        constructor(service: ProductsService) {
            // ...
            this.currentProduct = this.products[0];
        } 

        selectProduct(e) {
            this.currentProduct = e.itemData;
        }
    }

    <!-- tab: app.component.css -->
    #treeView, #product-details {
        display: inline-block;
        width: 300px;
    }

    #product-details {
        vertical-align: top;
        width: 400px;
        height: 420px;
        margin-left: 20px;
    }

    #product-details > img {
        border: none;
        height: 300px;
        width: 400px;
    }

    #product-details > .name {
        text-align: center;
        font-size: 20px;
    }

    #product-details > .price {
        text-align: center;
        font-size: 24px;
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <div>
            <DxTreeView ...
                selection-mode="single"
                :select-by-click="true"

>                 @item-selection-changed="selectProduct" 
>             >

Reviewer: ```suggestion
                @item-selection-changed="selectProduct">
```
